src/main/java/com/basho/riak/client/api/commands/ChunkedResponseIterator.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2016 Basho Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.basho.riak.client.api.commands;
+
+import com.basho.riak.client.core.StreamingRiakFuture;
+import com.basho.riak.client.core.util.BinaryValue;
+
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TransferQueue;
+import java.util.function.Function;
+
+public class ChunkedResponseIterator<FinalT, ChunkT extends Iterable<CoreT>, CoreT> implements Iterator<FinalT>
+{
+    private final int timeout;
+    private volatile BinaryValue continuation = null;
+    private final StreamingRiakFuture<ChunkT, ?> coreFuture;
+    private final TransferQueue<ChunkT> chunkQueue;
+    private final Function<CoreT, FinalT> createNext;
+    private final Function<ChunkT, Iterator<CoreT>> getNextIterator;
+    private final Function<ChunkT, BinaryValue> getContinuationFn;
```

Comment:
Is there a reason why it may require to expose BinaryValue as a part of getContinuationFn() signature?